Add 'dot' match option

webketje opened the pull request:
Manually integration-tested including loading from & saving to URL querystring

## Type of Change
<!-- What part of the source are you modifying? Remove the irrelevant options. -->

- **Tool Source:** Vue (app.vue), i18n
## What issue does this relate to?
Resolves #54 

### What should this PR do?
Adds "matchOptions" object to Vue app with (for the time being) a single option: dot (mirroring minimatch's).
Loads the option from & saves it to the URL querystring (as `options=dot:true`)
Also sets up foundation for adding support for other minimatch options

### What are the acceptance criteria?
Please check the translation.

| before | after |
|----------| --------|
|![Imgur](https://i.imgur.com/dzJCpvR.png) | ![Imgur](https://i.imgur.com/l9gjpNe.png) | |

MattIPv4 commented on src/glob-tool/templates/app.vue:
```diff
@@ -137,13 +145,19 @@ limitations under the License.
             commentsEnabled() {
                 this.test()
+            },
+            "matchOptions.dot": function() {
+                this.test()
             }
```
```suggestion
            matchOptions: {
                handler() {
                    this.test()
                },
                deep: true,
            },
```
To future-proof this if other settings are added, can we use a deep watcher for the whole object?

MattIPv4 commented on src/glob-tool/templates/app.vue:
```diff
@@ -183,6 +197,20 @@ limitations under the License.
+
+                options.forEach(o => {
+                    const [opt, value] = o.split(":")
+                    console.log(this.matchOptions)
```
```suggestion
```

webketje replied:
Oops.. forgot this

MattIPv4 reviewed (changes requested):
Ty for doing this -- two minor thoughts

webketje commented:
PS: before I commit fixes, any input regarding the translation?
Maybe it would be better to state "**wildcards** match path segments starting with a dot" (this is for all of `?`, `*`, and `**`)

MattIPv4 commented:
`Allow wildcards to match dotfiles?` maybe?

webketje commented:
> `Allow wildcards to match dotfiles?` maybe?

That is less complete though, as you can see in the "AFTER" screenshot it applies to any path segments with dots (eg also `.git/.nested/.env`)

webketje commented:
Ok I updated to latest master, changed translation to something I think is both clearer and more complete and adressed the issues you highlighted. Feel free to change the translation later if you find a better match :smile:

MattIPv4 commented:
@webketje what you have works for me -- would you mind rebasing against master again, I just merged a PR with the testing workflow marked as required here?

webketje commented:
Bump - done

MattIPv4 reviewed (approved):
LGTM, ty!
